File: backend/app/schemas/abuseipdb.py
from pydantic import BaseModel
from typing import Optional, List

class AbuseReport(BaseModel):
    reporterId: Optional[int]
    categories: List[int]
    reporterCountryCode: Optional[str]
    reportedAt: Optional[str]
    comment: Optional[str]

class IPReputationResult(BaseModel):
    ipAddress: str
    isPublic: bool
    ipVersion: int
    isWhitelisted: Optional[bool]
    abuseConfidenceScore: int   # 0-100, le score clé
    countryCode: Optional[str]
    usageType: Optional[str]
    isp: Optional[str]
    domain: Optional[str]
    totalReports: int
    numDistinctUsers: int
    lastReportedAt: Optional[str]
    reports: Optional[List[AbuseReport]]

File: frontend/src/types/abuseipdb.ts
export interface AbuseIPDBReport {
  reportedAt: string
  comment: string
  categories: number[]
  reporterId: number
  reporterCountryCode: string
  reporterCountryName: string
}

export interface AbuseIPDBResult {
  ipAddress: string
  isPublic: boolean
  ipVersion: number
  isWhitelisted: boolean | null
  abuseConfidenceScore: number
  countryCode: string | null
  countryName: string | null
  usageType: string | null
  isp: string | null
  domain: string | null
  hostnames: string[]
  isTor: boolean
  totalReports: number
  numDistinctUsers: number
  lastReportedAt: string | null
  reports: AbuseIPDBReport[]
}

export interface AbuseIPDBLookupResponse {
  ip: string
  result: AbuseIPDBResult
}